shimmingtoolbox/shim/sequencer.py

```diff
@@ -1371,44 +1387,102 @@ def plot_static_riro(self, masked_unshimmed, masked_shim_static, masked_shim_sta
         ax.set_title(f"shim static")
         ax = fig.add_subplot(2, 3, 6)
-        im = ax.imshow(np.rot90(unshimmed[..., i_slice, i_t]))
+        im = ax.imshow(np.rot90(unshimmed[..., i_slice, i_t]), vmin=min_value, vmax=max_value)
         fig.colorbar(im)
         ax.set_title(f"unshimmed")
         fname_figure = os.path.join(self.path_output, 'fig_realtime_masked_shimmed_vs_unshimmed.png')
         fig.savefig(fname_figure)
         logger.debug(f"Saved figure: {fname_figure}")
 
-    def plot_pressure_points(self, acq_pressures, ylim):
+    def plot_pressure_and_unshimmed_field(self, unshimmed_trace):
         """
-        Plot respiratory trace pressure points
+        Plot respiratory trace, acquisition time pressure points and the B0 field RMSE
 
         Args:
-            acq_pressures (np.ndarray): acquisitions pressures
-            ylim (float): Limit of the y-axis
-
-        """
-        fig = Figure(figsize=(10, 10))
-        ax = fig.add_subplot(111)
-        ax.plot(acq_pressures, label='pressures')
-        ax.legend()
-        ax.set_ylim(ylim)
-        ax.set_title("Pressures vs time points")
-        fname_figure = os.path.join(self.path_output, 'fig_trace_pressures.png')
-        fig.savefig(fname_figure)
+            unshimmed_trace (np.ndarray): field in the ROI for each shim volume
+        """
+        # Get the pmu data values in the range of the acquisition
+        pmu_timestamps = self.pmu.get_times()
+        pmu_pressures = self.pmu.data
+        indexes = np.where(np.logical_and(pmu_timestamps >= (self.acq_timestamps[0] - 1000),
+                                          pmu_timestamps <= self.acq_timestamps[-1] + 1000))
+        pmu_timestamps_curated = pmu_timestamps[indexes]
+        pmu_pressures_curated = pmu_pressures[indexes]
+
+        # Select slices shimmed
+        curated_unshimmed_trace = unshimmed_trace[self.index_shimmed]
+
+        # Get the b0 field in the same units as the pressure reading
+        n_plots = len(self.index_shimmed)
+        max_diff_field = 0
+        for i_plot in range(n_plots):
+            diff_field = curated_unshimmed_trace[i_plot].max() - curated_unshimmed_trace[i_plot].min()
+            if abs(max_diff_field) < abs(diff_field):
+                max_diff_field = diff_field
+                min_field = curated_unshimmed_trace[i_plot].min()
+                max_field = curated_unshimmed_trace[i_plot].max()
+
+        diff_pressure = pmu_pressures_curated.max() - pmu_pressures_curated.min()
+        scaling = max_diff_field / diff_pressure
+        avg_pressure = np.mean(pmu_pressures_curated)
+
+        curated_unshimmed_trace_scaled = np.zeros_like(curated_unshimmed_trace)
+        for i_plot in range(n_plots):
+            avg_b0field = np.mean(curated_unshimmed_trace[i_plot])
+            curated_unshimmed_trace_scaled[i_plot] = (curated_unshimmed_trace[
+                                                          i_plot] - avg_b0field) / scaling + avg_pressure
```

Comment:
```suggestion
        curated_unshimmed_trace_scaled = np.array([(x - np.max(x)) / scaling + avg_pressure for x in curated_unshimmed_trace])
```

Reply:
depends if n_plot is the same length as curated_unshimmed_trace

Reply:
It is, I used np.mean but otherwise changed it to your suggestion in: [b8f20da](https://github.com/shimming-toolbox/shimming-toolbox/pull/462/commits/b8f20da1744efa41af3bf649551277518322d6a8)